src/microbe_stage/components/MicrobeControl.cs

```diff
@@ -195,4 +195,55 @@ public static void SetMoveSpeed(this ref MicrobeControl control, float speed)
             control.QueuedSlimeSecretionTime = 3 * random.NextSingle();
         }
     }
+
+    public static void SetMucocystState(this ref MicrobeControl control,
+        ref OrganelleContainer organelleInfo, in Entity entity, bool state)
+    {
+        if (entity.Has<MicrobeColony>())
+        {
+            ref var colony = ref entity.Get<MicrobeColony>();
+
+            // TODO: is it a good idea to allocate a delegate here?
+            colony.PerformForOtherColonyMembersThanLeader(m =>
+                m.Get<MicrobeControl>()
+                    .SetMucocystState(ref m.Get<OrganelleContainer>(), m, state));
+        }
+
+        // It might be tricky as there is now mucocyst upgrade that doesn't emit slime normally
+        if (organelleInfo.Mucocysts == null || organelleInfo.Mucocysts.Count < 1)
+            return;
+
+        // Must have at least one mucocyst
+        if (organelleInfo.Mucocysts?.Count < 1)
+            return;
+
+        if (entity.Has<CompoundStorage>())
+        {
+            ref var compoundStorage = ref entity.Get<CompoundStorage>();
+            var mucilage = SimulationParameters.Instance.GetCompound("mucilage");
+
+            if (compoundStorage.Compounds.Compounds[mucilage] < Constants.MUCOCYST_MINIMUM_MUCILAGE)
+                return;
+        }
+
+        if (state)
+        {
+            control.State = MicrobeState.MucocystShield;
+            var membrane = entity.Get<CellProperties>().CreatedMembrane;
+
+            membrane?.SetMucocystEffectVisible(true);
```

Comment:
This method should not directly modify the graphics like this, this goes against the ECS architecture. Instead this should just let the mucocyst system handle the graphics.

Reply:
I would need help with this

Reply:
Just remove the code from here, as far as I remember reading the code the mucocyst system should already be setup to apply the visual effect to the membrane.